src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs

```diff
@@ -32,172 +24,72 @@ internal sealed class DangerousMemberUsagesAnalyzer : DiagnosticAnalyzer {
 
 		private void RegisterAnalysis( CompilationStartAnalysisContext context ) {
 
-			RegisterDangerousMethodAnalysis( context );
-			RegisterDangerousPropertyAnalysis( context );
-		}
-
-		private void RegisterDangerousMethodAnalysis( CompilationStartAnalysisContext context ) {
-
 			Compilation compilation = context.Compilation;
 
-			INamedTypeSymbol auditedAttributeType = compilation.GetTypeByMetadataName( DangerousMethodAuditedAttributeFullName );
-			INamedTypeSymbol unauditedAttributeType = compilation.GetTypeByMetadataName( DangerousMethodUnauditedAttributeFullName );
-			IImmutableSet<ISymbol> dangerousMethods = GetDangerousMethods( compilation );
-
-			context.RegisterSyntaxNodeAction(
-					ctxt => {
-						if( ctxt.Node is InvocationExpressionSyntax invocation ) {
-							AnalyzeMethodInvocation( ctxt, invocation, auditedAttributeType, unauditedAttributeType, dangerousMethods );
-						}
-					},
-					SyntaxKind.InvocationExpression
-				);
-		}
-
-		private void RegisterDangerousPropertyAnalysis( CompilationStartAnalysisContext context ) {
-
-			Compilation compilation = context.Compilation;
-
-			INamedTypeSymbol auditedAttributeType = compilation.GetTypeByMetadataName( DangerousPropertyAuditedAttributeFullName );
-			INamedTypeSymbol unauditedAttributeType = compilation.GetTypeByMetadataName( DangerousPropertyUnauditedAttributeFullName );
-			IImmutableSet<ISymbol> dangerousProperties = GetDangerousProperties( compilation );
-
-			context.RegisterSyntaxNodeAction(
-					ctxt => {
-						if( ctxt.Node is MemberAccessExpressionSyntax propertyAccess ) {
-							AnalyzePropertyAccess( ctxt, propertyAccess, auditedAttributeType, unauditedAttributeType, dangerousProperties );
-						}
-					},
-					SyntaxKind.SimpleMemberAccessExpression
-				);
-		}
-
-		private void AnalyzeMethodInvocation(

```

Comment:
Simplified and moved other stuff to the model

![image](https://user-images.githubusercontent.com/12101120/151664172-a486c971-359d-4001-81aa-10402da5c85a.png)